components/product/list.js

```diff
@@ -1,15 +1,45 @@
 import { SingleProductComponent } from "./single";
-
-export const ListProductComponent = ({ products = [], handleAddToCartClick }) => {
+import React, { useState } from "react";
+export const ListProductComponent = ({
+  products = [],
+  handleAddToCartClick,
+}) => {
+  const [sortProduct, setSortProduct] = useState(false);
+  const sortByPrice = () => {
+    setSortProduct(!sortProduct);
+  };
+  const compare = (a, b) => {
```

Comment:
functions like this, that don't have dependency on any state variable, should be defined outside of the component scope.

Reply:
Got it. Thanks!